Script/update_grovepi.sh

```diff
@@ -5,34 +5,147 @@ PIHOME=/home/pi
+GROVEPI_DIR=$DEXTER_PATH/GrovePi
+DEXTERSCRIPT=$DEXTER_PATH/lib/Dexter/script_tools
 
-# needs to be sourced from here when we call this as a standalone
-source /home/pi/$DEXTER/lib/$DEXTER/script_tools/functions_library.sh
+# whether to install the dependencies or not (avrdude, apt-get, wiringpi, and so on)
+installdependencies=true
+updaterepo=true
+install_pkg_scriptools=true
 
-GROVEPI_DIR=$DEXTER_PATH/GrovePi
+# the following 3 options are mutually exclusive
+systemwide=true
+userlocal=false
+envlocal=false
+usepython3exec=true
+
+# the following option tells which branch has to be used
+selectedbranch="master"
 
-# Check for a GrovePi directory under "Dexter" folder.  If it doesn't exist, create it.
+declare -a optionslist=("--system-wide")
+# iterate through bash arguments
+for i; do
+  case "$i" in
+    --no-dependencies)
+      installdependencies=false
+      ;;
+    --no-update-aptget)
+      updaterepo=false
+      ;;
+    --bypass-pkg-scriptools)
+      install_pkg_scriptools=false
+      ;;
+    --user-local)
+      userlocal=true
+      systemwide=false
+      declare -a optionslist=("--user-local")
+      ;;
+    --env-local)
+      envlocal=true
+      systemwide=false
+      declare -a optionslist=("--env-local")
+      ;;
+    --system-wide)
+      ;;
+    develop|feature/*|hotfix/*|fix/*|DexterOS*|v*)
+      selectedbranch="$i"
+      ;;
+  esac
+done
 
-if [ -d "$GROVEPI_DIR" ]; then
-    echo "GrovePi Directory Exists"
-    cd $GROVEPI_DIR             # Go to directory
-    sudo git fetch origin       # Hard reset the git files
-    sudo git reset --hard  
-    sudo git merge origin/master
+# show some feedback for the GrovePi
+echo "  _____            _                                ";
+echo " |  __ \          | |                               ";
+echo " | |  | | _____  _| |_ ___ _ __                     ";
+echo " | |  | |/ _ \ \/ / __/ _ \ '__|                    ";
+echo " | |__| |  __/>  <| ||  __/ |                       ";
+echo " |_____/ \___/_/\_\\\__\___|_|          _            ";
+echo " |_   _|         | |         | |      (_)           ";
+echo "   | |  _ __   __| |_   _ ___| |_ _ __ _  ___  ___  ";
+echo "   | | | '_ \ / _\ | | | / __| __| '__| |/ _ \/ __| ";
+echo "  _| |_| | | | (_| | |_| \__ \ |_| |  | |  __/\__ \ ";
+echo " |_____|_| |_|\__,_|\__,_|___/\__|_|  |_|\___||___/ ";
+echo "                                                    ";
+echo "                                                    ";
+echo "  _____                    _____ _ "
+echo " / ____|                  |  __ (_)  "
+echo "| |  __ _ __ _____   _____| |__) |   "
+echo "| | |_ | '__/ _ \ \ / / _ \  ___/ |  "
+echo "| |__| | | | (_) \ V /  __/ |   | |  "
+echo " \_____|_|  \___/ \_/ \___|_|   |_|  "
+echo " "
+
+# show some feedback on the console
+if [ -f $DEXTERSCRIPT/functions_library.sh ]; then
+  source $DEXTERSCRIPT/functions_library.sh
```

Comment:
This is silly, either functions_library.sh is used or it's not.   Pull it in, curl it if necessary, and make use of it.

Reply:
Since `install.sh` is a big mess, I had to containerize it and make it work like a module that only installs dependencies and I moved the rest here.
Above that, we have to first parse the arguments and right after that we can curl the command. We can't curl it twice for this matter, it doesn't make sense. We can't curl it and then parse the arguments. So, instead we check if, `script_tools` is installed, and if it is, we use the `feedback` function, else we just use a regular echo. This means that `feedback` will work only if a certain robot has already been installed once, and that's fine.

Reply:
install.sh is what is referenced in tutorials all over the place as being the proper way to install GrovePi. 
We have to live with that. You can't just go in and move stuff from one script to another.

Reply:
I can't contest the presence of `install.sh` in other tutorials.

How do you envision the solution?